pkg/controllers/management/clusterupstreamrefresher/eks_upstream_spec.go

```diff
@@ -0,0 +1,137 @@
+	sess, eksService, err := ekscontroller.StartAWSSessions(secretsCache, *cluster.Spec.EKSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterState, err := eksService.DescribeCluster(
+		&eks.DescribeClusterInput{
+			Name: aws.String(cluster.Spec.EKSConfig.DisplayName),
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	ngs, err := eksService.ListNodegroups(
+		&eks.ListNodegroupsInput{
+			ClusterName: aws.String(cluster.Spec.EKSConfig.DisplayName),
+		})
+
+	// gather upstream node groups states
+	var nodeGroupStates []*eks.DescribeNodegroupOutput
+	var errs []string
+	for _, ngName := range ngs.Nodegroups {
+		ng, err := eksService.DescribeNodegroup(
+			&eks.DescribeNodegroupInput{
+				ClusterName:   aws.String(cluster.Spec.EKSConfig.DisplayName),
+				NodegroupName: ngName,
+			})
+		if err != nil {
+			return nil, err
+		}
+
+		nodeGroupStates = append(nodeGroupStates, ng)
+		var nodeGroupMustBeDeleted string
+		if len(ng.Nodegroup.Health.Issues) != 0 {
+			var issueMessages []string
+			for _, issue := range ng.Nodegroup.Health.Issues {
+				issueMessages = append(issueMessages, aws.StringValue(issue.Message))
+				if !ekscontroller.NodeGroupIssueIsUpdatable(aws.StringValue(issue.Code)) {
+					nodeGroupMustBeDeleted = ": node group cannot be updated, must be deleted and recreated"
+				}
+			}
+			errs = append(errs, fmt.Sprintf("health error for node group [%s] in cluster [%s]: %s%s",
+				aws.StringValue(ng.Nodegroup.NodegroupName),
+				cluster.Name,
+				strings.Join(issueMessages, "; "),
+				nodeGroupMustBeDeleted,
+			))
+		}
+	}
+
+	upstreamSpec, _, err := ekscontroller.BuildUpstreamClusterState(cluster.Spec.DisplayName, cluster.Status.EKSStatus.ManagedLaunchTemplateID, clusterState, nodeGroupStates, ec2.New(sess), false)
+	if err != nil {
+		// If we get an error here, then syncing is broken
+		return nil, err
+	}
+
+	upstreamSpec.DisplayName = cluster.Spec.EKSConfig.DisplayName
+	upstreamSpec.Region = cluster.Spec.EKSConfig.Region
+	upstreamSpec.AmazonCredentialSecret = cluster.Spec.EKSConfig.AmazonCredentialSecret
+	upstreamSpec.Imported = cluster.Spec.EKSConfig.Imported
+	upstreamSpec.Subnets = cluster.Spec.EKSConfig.Subnets
+	upstreamSpec.SecurityGroups = cluster.Spec.EKSConfig.SecurityGroups
+	upstreamSpec.ServiceRole = cluster.Spec.EKSConfig.ServiceRole
+
+	if len(errs) != 0 {
+		// If there are errors here, we can still sync, but there are problems with the nodegroups that should be reported
+		err = fmt.Errorf("error for cluster [%s]: %s",
+			cluster.Name,
+			strings.Join(errs, "\n"))
+	}
+
+	return upstreamSpec, err
+}
+
+// MigrateEksRefreshCronSetting migrate deprecated EKS refresh cron setting to new format
+// and remove deprecated eks-refresh-cron setting object, if setting is not found then return
+// This function will be run only once in pkg/multiclustermanager/app.go
+func MigrateEksRefreshCronSetting(wContext *wrangler.Context) error {
+	settingsClient := wContext.Mgmt.Setting()
```

Comment:
As I understand it we can't do it this way because the provider (storage) isn't used. We can do how `StartEKSUpstreamCronJob` is done today, just import settings package and check there. Though interestingly we can't get just the Value from there. So I suppose we will have to check if `Get()` and `Default` are different and then if they are we assume its set, and if they are not we assume its default.

Reply:
Actually I think I am wrong here

Reply:
After bringing back `EKSUpstreamRefreshCron            = NewSetting("eks-refresh-cron", "*/5 * * * *")` to `pkg/settings/setting.go` we can get setting value in both ways.

Reply:
According to our discussion on Slack it stayed as it is